community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/RecoveryVisitor.java

```diff
@@ -54,11 +62,14 @@ public RecoveryVisitor( TransactionIdStore store,
     public boolean visit( CommittedTransactionRepresentation transaction ) throws IOException
     {
         long txId = transaction.getCommitEntry().getTxId();
-        try ( LockGroup locks = new LockGroup() )
+        TransactionRepresentation txRepresentation = transaction.getTransactionRepresentation();
+
+        try ( LockGroup locks = new LockGroup();
+              ValidatedIndexUpdates indexUpdates = indexUpdatesValidator.validate( txRepresentation, RECOVERY ) )
```

Comment:
First when reading this I found it strange that we need to validate the index capacity constraint at recovery knowing that we wouldn't put offending transactions in the log to start with. But I'd guess it due to the fact that it's not _just_ validation, it's also the grouping of the commands... something that is assumed to be done and passed into the store applier. Is that the case? Could you clarify that by a comment perhaps?